### PR TITLE
Add ipv6 vips reservation

### PR DESCRIPTION
**Pull request by jadhaj**
The goal of this PR is to Add ipv6 vips reservation to reserve-hosts.sh. This is needed for dual stack & single stack IPv6 deployments

### CODE REVIEW

File: images/fcos-bastion-image/root/usr/bin/reserve-hosts.sh [hunk 1]
```diff
@@ -119,6 +119,7 @@ printf '%s\n' "${OUTPUT[@]}" >> "$RESERVED_FILE"
 if [ x"$REQUEST_VIPS" == x"true" ]; then
     FIRST_HOST_ID=$(echo "${CANDIDATES_HOSTS[0]}" | cut -f3 -d,)
     echo "{\"ingress_vip\": \"${INGRESS_VIPS_SUBNET_PREFIX:-192.168.82.}${FIRST_HOST_ID}\", \"api_vip\": \"${API_VIPS_SUBNET_PREFIX:-192.168.81.}${FIRST_HOST_ID}\"}" > "${BUILD_DIR}/vips.json"
+    echo "{\"ingress_vip_v6\": \"${IPV6_INGRESS_VIPS_SUBNET_PREFIX:-fd99:2222:3456::2:}${FIRST_HOST_ID}\", \"api_vip_v6\": \"${IPV6_API_VIPS_SUBNET_PREFIX:-fd99:2222:3456::1:}${FIRST_HOST_ID}\"}" >> "${BUILD_DIR}/vips.json"
```
Comment (aleskandro) [inline images/fcos-bastion-image/root/usr/bin/reserve-hosts.sh:122]: FIRST_HOST_ID can be either 1, 2, or 3 characters. Can we print it as an hexadecimal representation and force it to be `00$FIRST_HOST_ID_HEX`?

Comment (aleskandro) [reply images/fcos-bastion-image/root/usr/bin/reserve-hosts.sh:122]: Also, you should add the two keys in the row above as you are printing two json maps into a plain json file... which is an invalid json

Comment (jadhaj) [reply images/fcos-bastion-image/root/usr/bin/reserve-hosts.sh:122]: yes, make sense, they should combined on one row
regardless, there is a lot of shellcheck failures not related to my changes

Comment (jadhaj) [reply images/fcos-bastion-image/root/usr/bin/reserve-hosts.sh]: Done

Comment (jadhaj) [reply images/fcos-bastion-image/root/usr/bin/reserve-hosts.sh]: shellcheck is complaining about:
Avoid x-prefix in comparisons as it no longer serves a purpose.

should I remove these ?

Comment (jadhaj) [reply images/fcos-bastion-image/root/usr/bin/reserve-hosts.sh]: Shellcheck job passed after removing x-prefix and unused var, but images still failing

Comment (jadhaj) [reply images/fcos-bastion-image/root/usr/bin/reserve-hosts.sh]: All checks passed

**Comment by jadhaj** [comment]
@aleskandro anything else needed here ?

File: images/fcos-bastion-image/root/usr/bin/reserve-hosts.sh [hunk 2]
```diff
@@ -18,6 +18,7 @@ VENDOR_REGEX=""
 
 if [ -n "$VENDOR" ]; then
   VENDOR_REGEX=",${VENDOR},"
+  # shellcheck disable=SC2086
   echo $VENDOR_REGEX
```
Comment (aleskandro) [inline images/fcos-bastion-image/root/usr/bin/reserve-hosts.sh:22]:
```suggestion
```

Let's just remove this line. @bmanzari do you confirm it was added only for debugging purposes?

Comment (jadhaj) [reply images/fcos-bastion-image/root/usr/bin/reserve-hosts.sh]: I run below script for testing:

cat test_vips.sh
#!/bin/bash

REQUEST_VIPS="true"
CANDIDATES_HOSTS=("host1,host2,1" "host3,host4,10" "host5,host6,255")
BUILD_DIR="."

# Script logic
if [ "$REQUEST_VIPS" == "true" ]; then
    FIRST_HOST_ID=$(echo "${CANDIDATES_HOSTS[0]}" | cut -f3 -d,)
    FIRST_HOST_ID_HEX=$(printf '%03x' "$FIRST_HOST_ID")

    INGRESS_VIP="${INGRESS_VIPS_SUBNET_PREFIX:-192.168.82.}${FIRST_HOST_ID}"
    API_VIP="${API_VIPS_SUBNET_PREFIX:-192.168.81.}${FIRST_HOST_ID}"
    INGRESS_VIP_V6="${IPV6_INGRESS_VIPS_SUBNET_PREFIX:-fd99:2222:3456::2:}${FIRST_HOST_ID_HEX}"
    API_VIP_V6="${IPV6_API_VIPS_SUBNET_PREFIX:-fd99:2222:3456::1:}${FIRST_HOST_ID_HEX}"

    # Create a single JSON object
    echo "{
  \"ingress_vip\": \"$INGRESS_VIP\",
  \"api_vip\": \"$API_VIP\",
  \"ingress_vip_v6\": \"$INGRESS_VIP_V6\",
  \"api_vip_v6\": \"$API_VIP_V6\"
}" > "${BUILD_DIR}/vips.json"
fi

The result is:
cat vips.json
{
  "ingress_vip": "192.168.82.1",
  "api_vip": "192.168.81.1",
  "ingress_vip_v6": "fd99:2222:3456::2:001",
  "api_vip_v6": "fd99:2222:3456::1:001"
}

**Pull request merged**